js/state/frameStateUtil.js

```diff
@@ -460,15 +460,17 @@ function isPinned (state, frameKey) {
 function updateTabPageIndex (state, frameProps) {
   frameProps = makeImmutable(frameProps)
   const index = getFrameTabPageIndex(state, frameProps.get('key'))
+  const isTabInHoverState = !!getHoverTabIndex(state)
 
   if (index === -1) {
     return state
   }
 
-  state = state.setIn(['ui', 'tabs', 'tabPageIndex'], index)
-  state = state.deleteIn(['ui', 'tabs', 'previewTabPageIndex'])
-
-  return state
+  // Do not update tabPageIndex if user is in hover mode
```

Comment:
The concern here is that updateTabPageIndex should just do what it says and not ignore for `isTabInHoverState`.

Instead we should just call `updateTabPageIndex` when the active state of a tab changes. 
Because otherwise you could be on tab page 2 with an active tab on tab page 3. And as you're looking at tab page 2, audio could start playing on tab page 3 and it would change the active tab. 

What you did mostly fixes it, but there are some edge cases that are missed. 

I posted this issue for tracking the work, I'll merge this in the meantime:
https://github.com/brave/browser-laptop/issues/9911

You can do that task out of line with lower priority.